include/oneapi/dpl/internal/async_extension_impl.h

```diff
@@ -0,0 +1,556 @@
+/*
+ *  Copyright (c) 2020 Intel Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#ifndef _ONEDPL_ASYNC_EXTENSION_IMPL_H
+#define _ONEDPL_ASYNC_EXTENSION_IMPL_H
+
+namespace oneapi
+{
+namespace dpl
+{
+namespace __par_backend_hetero
+{
+
+//------------------------------------------------------------------------
+// parallel_transform_reduce - async pattern 2.0
```

Comment:
Remove the version number.  Should these be moved into include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h so they're in the same location as their sync counterparts and other async building blocks?